src/server/journal/types.h

```diff
@@ -87,7 +87,12 @@ struct JournalItem {
   std::optional<SlotId> slot;
 };
 
-using ChangeCallback = std::function<void(const JournalItem&, bool await)>;
+struct JournalConsumerInterface {
+  virtual ~JournalConsumerInterface() = default;
+
+  virtual void ConsumeJournalChange(const JournalItem& item) = 0;
```

Comment:
I see you liked the proposal :)

Please add comments to when each method is called.

Reply:
The proposal was good, I wanted to see we pass all tests with it.
There is some down side to it though, the snapshot.cc is also a consumer of journal changes and we do not have an async socket write in this flow. This means that when calling ConsumeJournalChange in this flow we serialize the data and when calling Throttle we actually flush to sync the serialized data. This also means that if we have few snapshot consumers we first serialize on all and than flush on all, before this change we serialized and flushed if needed on each snapshot instance

Reply:
correct. what happens if we have big values in both consumers in the same loop? will it cause trouble/deadlock?

Reply:
ah, since it's journaling, big values are not related.